dlib/dnn/layers.h

```diff
@@ -2241,12 +2241,14 @@ namespace dlib
             if (version != "add_prev_")
                 throw serialization_error("Unexpected version '"+version+"' found while deserializing dlib::add_prev_.");
         }
-
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wunused-parameter"
```

Comment:
This is a pretty heavy solution.  How about just saying `friend std::ostream& operator<<(std::ostream& out, const add_prev_& )`?

Reply:
Yes, you're right, that works better :) thank you!